float32[6] mic_energy
float32 angle
float32 mag